server/src/main/java/org/elasticsearch/cluster/block/ClusterBlockException.java

```diff
@@ -110,12 +111,18 @@ private static String buildMessageForIndexBlocks(Map<String, Set<ClusterBlock>>
     @Override
     public RestStatus status() {
         RestStatus status = null;
+        boolean onlyRetryableBlocks = false;
         for (ClusterBlock block : blocks) {
             if (status == null) {
                 status = block.status();
             } else if (status.getStatus() < block.status().getStatus()) {
                 status = block.status();
             }
+            onlyRetryableBlocks = block.id() == IndexMetaData.INDEX_READ_ONLY_ALLOW_DELETE_BLOCK.id();
```

Comment:
I would prefer to compare `block.status() != RestStatus.TOO_MANY_REQUESTS` here.

Reply:
I have changed the code yet.